AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -560,6 +560,28 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)__setNeedsLayout
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (!_flags.isMeasured) {
+    return;
+  }
+  
+  ASSizeRange oldConstrainedSize = _constrainedSize;
+  [self invalidateCalculatedLayout];
+  
+  if (_supernode) {
+    // Cause supernode's layout to be invalidated
+    [_supernode setNeedsLayout];
+  } else {
+    // This is the root node. Trigger a full measurement pass on *current* thread. Old constrained size is re-used.
+    [self measureWithSizeRange:oldConstrainedSize];
+    self.frame = CGRectMake(0.0f, 0.0f, _layout.size.width, _layout.size.height);
```

Comment:
Nice catch on the bounds setter. I didn't consider it!

Indeed, view/layer -setNeedsLayout is called as always, in [ASDisplayNode+UIViewBridge.mm](https://github.com/facebook/AsyncDisplayKit/pull/594/files#diff-4c96d795186d237f899d33db7bbf4a1aL174).

What do you mean by "then -setNeedsLayout should be set in the pending state"? Do we still need it?